data/src/main/java/me/androidbox/data/alarm_manager/AlarmReceiver.kt

```diff
@@ -1,15 +1,75 @@
 package me.androidbox.data.alarm_manager
 
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
 import android.content.BroadcastReceiver
 import android.content.Context
 import android.content.Intent
+import android.os.Build
+import androidx.core.app.NotificationCompat
+import me.androidbox.data.R
+import me.androidbox.data.alarm_manager.AlarmSchedulerImp.Companion.EXTRA_AGENDA_TYPE
+import me.androidbox.data.alarm_manager.AlarmSchedulerImp.Companion.EXTRA_DESCRIPTION
+import me.androidbox.data.alarm_manager.AlarmSchedulerImp.Companion.EXTRA_ID
+import me.androidbox.data.alarm_manager.AlarmSchedulerImp.Companion.EXTRA_TITLE
 
 class AlarmReceiver : BroadcastReceiver() {
 
     override fun onReceive(context: Context?, intent: Intent?) {
-        intent?.getStringExtra("EXTRA_MESSAGE")?.let { message ->
-            /* TODO Add Notification here */
-            println("Alarm triggered: $message")
+        /* TODO Not sure if intent and context can be null but I guess is ok to cast it anyway */
+        if(intent != null && context != null) {
```

Comment:
@MKiperszmid @philipplackner Do we need to check for null context and intent. Would they really be null?

Reply:
I never had a null intent and context, but I prefer checking them just in case


Here to make it a bit more clearer I like to do:

if(intent == null || context == null)  return


So you just add that at the top, and done
You don't need to put everything inside an if

Reply:
I think technically it could happen if whatever sends the broadcast (here AlarmManager) doesn't attach anything, but for AlarmManager you can be pretty sure that doesn't happen